### DISCUSSION TITLE
Pass transaction object in all available methods

### DISCUSSION THREAD

**Discussion by klesgidis** [timgit/pg-boss#338] [Ideas]
Hey guys,

firstly thanks for the great work that you've done!

We're currently evaluating `pg-boss` as a queueing system in our product and I'd like to ask if you are thinking of adding transaction functionality as an option for all of your methods. 

We'd like to call methods like `sendJob` inside a transaction & if the transaction fails the job should be skipped. We're currently directly insert into the DB the payload for the job, but this is not maintainable in production and we have to re-implement all the methods implemented by you already.

Thanks a lot,
Kyriakos

**Comment by timgit** [comment]
Yes, there was previously a PR (https://github.com/timgit/pg-boss/pull/200) that was almost merged to do this.  It's incompatible to merge at this point, but the comment history offers good context for what a new PR would look like.

**Comment by klesgidis** [comment]
@timgit thanks for the answer. I am trying to open a PR but I don't have the required permissions

**Reply by klesgidis** [reply]
Hey @timgit any update on the permissions?! Thanks

**Reply by timgit** [reply]
It's a public repo. You should be able to fork and open a PR.

**Reply by klesgidis** [reply]
Thank you!

**Comment by klesgidis** [comment]
https://github.com/timgit/pg-boss/pull/341

**Reply by timgit** [reply]
Merged
